crates/autopilot/src/domain/competition/mod.rs

```diff
@@ -0,0 +1,66 @@
+use {
+    super::{auction, eth},
+    crate::domain,
+    number::nonzero::U256 as NonZeroU256,
+    std::collections::HashMap,
+};
+
+type SolutionId = u64;
+
+pub struct Solution {
+    id: SolutionId,
+    account: eth::Address,
+    score: NonZeroU256,
```

Comment:
I think we should follow the practices established in other domains and have a new type for `SolutionId` and `Score` since they are quite important concepts IMO.

Reply:
Covered in https://github.com/cowprotocol/services/pull/2746